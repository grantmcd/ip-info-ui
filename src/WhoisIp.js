import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { Card, ListGroup, Alert, Accordion, Button } from "react-bootstrap";

const WhoisIp = (props) => {
  const [hasError, setErrors] = useState(false);
  const [whoisData, setWhoisData] = useState(null);

  useEffect(() => {
    async function fetchData(ipAddress) {
      const geoIpRes = await fetch(
        "https://hu8qooiisa.execute-api.us-east-2.amazonaws.com/dev/whois-ip/" +
          props.ipAddress
      );

      geoIpRes
        .json()
        .then((res) => {
          setWhoisData(res);
        })
        .catch((err) => setErrors(err));
    }

    if (props.ipAddress) {
      fetchData(props.ipAddress);
    }
  }, [props.ipAddress]);

  return (
    <Card className="mb-3">
      <Card.Header>Whois</Card.Header>
      {hasError && (
        <Card.Body>
          <Alert variant="warning">Something went wrong: {hasError}</Alert>
        </Card.Body>
      )}

      {whoisData && (
        <ListGroup variant="flush">
          <ListGroup.Item>
            Organization: {whoisData.Organization}
          </ListGroup.Item>
          <ListGroup.Item>
            Registration Date: {whoisData.RegDate}
          </ListGroup.Item>

          {/* <ListGroup.Item>City: {geoData.city.names.en}</ListGroup.Item>
          <ListGroup.Item>Latitude: {geoData.location.latitude}</ListGroup.Item>
          <ListGroup.Item>
            Longitude: {geoData.location.longitude}
          </ListGroup.Item> */}
        </ListGroup>
      )}
    </Card>
  );
};

WhoisIp.propTypes = {
  ipAddress: PropTypes.string.isRequired,
};

export default WhoisIp;
