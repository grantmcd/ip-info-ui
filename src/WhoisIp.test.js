import React from "react";
import WhoisIp from "./WhoisIp";
import renderer from "react-test-renderer";

test("GeoIp when Ip Address is set", () => {
  const component = renderer.create(<WhoisIp ipAddress="75.34.225.8" />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
