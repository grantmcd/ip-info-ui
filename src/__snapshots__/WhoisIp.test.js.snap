// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoIp when Ip Address is set 1`] = `
<div
  className="mb-3 card"
>
  <div
    className="card-header"
  >
    Whois
  </div>
</div>
`;
